pdf-generator/src/lib/services/FileService.ts

```
@@ -0,0 +1,291 @@
+  }
+
+  /**
+   * Check for suspicious content in filename
+   */
+  private containsSuspiciousContent(filename: string): boolean {
+    const suspiciousPatterns = [
+      /\.\./, // Directory traversal
+      /[<>:"|?*]/, // Invalid Windows characters
+      /^(CON|PRN|AUX|NUL|COM[1-9]|LPT[1-9])$/i, // Windows reserved names
+      /^\./, // Hidden files starting with dot
+      /\x00/, // Null bytes
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)